functions/src/sections/Section.service.ts

```diff
@@ -1,123 +1,129 @@
+      })
+    );
   }
+  return [];
+}
 
-  public async getSectionMapping(
-    term: string,
-    subject: string,
-    code: string
-  ): Promise<CourseMapping | undefined> {
-    const doc = await db.courseMappings
-      .doc(SectionsService.constructSectionKey(term, subject, code))
-      .get();
+export async function getSectionMapping(
+  term: string,
+  subject: string,
+  code: string
+): Promise<CourseMapping | undefined> {
+  const doc = await get(
+    collection.courseMappings,
+    constructSectionKey(term, subject, code)
+  );
 
-    // FIX: serialization to and from firestore isn't working as expected
-    // disable document staleness check for now.
-    // const t = doc.data()?.retrievedAt?.getTime();
-    // // if retrievedAt exists and it wasn't retieved within 30 minutes
-    // if (t && t + 1000 * 1800 > Date.now()) {
-    //   return undefined;
-    // }
-    return doc.data();
-  }
+  // FIX: serialization to and from firestore isn't working as expected
+  // disable document staleness check for now.
+  // const t = doc.data()?.retrievedAt?.getTime();
+  // // if retrievedAt exists and it wasn't retieved within 30 minutes
+  // if (t && t + 1000 * 1800 > Date.now()) {
+  //   return undefined;
+  // }
+  return doc?.data;
+}
 
-  public async setSectionMapping(
-    term: string,
-    subject: string,
-    code: string
-  ): Promise<CourseMapping | void> {
-    try {
-      const sections = await UVicCourseScraper.getCourseSections(
-        term,
-        subject,
-        code
+export async function setSectionMapping(
+  term: string,
+  subject: string,
+  code: string
+): Promise<CourseMapping | void> {
+  try {
+    const sections = await UVicCourseScraper.getCourseSections(
+      term,
+      subject,
+      code
+    );
+
+    if (sections.length > 0) {
+      const crns = sections.map(({ crn }) => crn);
+      const retrievedAt = new Date(Date.now());
+      await set(
+        collection.courseMappings,
+        constructSectionKey(term, subject, code),
+        { crns, retrievedAt }
       );
-      if (sections.length > 0) {
-        const crns = sections.map(({ crn }) => crn);
-        const retrievedAt = new Date(Date.now());
-        await db.courseMappings
-          .doc(SectionsService.constructSectionKey(term, subject, code))
-          .set({ crns, retrievedAt });
-        return { crns, retrievedAt };
-      }
-    } catch (e) {
-      console.warn(e);
+
+      return { crns, retrievedAt };
     }
+  } catch (e) {
+    console.warn(e);
   }
+}
 
-  public async updateSectionMappings(): Promise<void> {
-    // TODO: a good candidate for using RemoteConfig :)
-    const terms = ['202009'];
-    // get all the courses
-    const courses = (await UVicCourseScraper.getCourses()).map((course) =>
-      subjectCodeExtractor(course)
-    );
+export async function updateSectionMappings(): Promise<void> {
+  // TODO: a good candidate for using RemoteConfig :)
+  const terms = ['202009'];
```

Comment:
I know there's a TODO here but is there a reason this term is hard coded in?

Reply:
The diff shows this but I didn't do it in this PR.

Reply:
nothing calls this endpoint right now (im gonna replace it soon anyhow :shrug: )